core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/DataAccessImpl.java

```diff
@@ -755,53 +967,110 @@ protected void initPreparedStatements() {
                 });
     }
 
+    Long tableToMapKey(String tableName) {
+        LocalDateTime parsed = LocalDateTime
+                .parse(tableName.substring(TEMP_TABLE_NAME_PROTOTYPE.length()),
+                        TEMP_TABLE_DATEFORMATTER);
+        return Long.valueOf(parsed.toInstant(ZoneOffset.UTC).toEpochMilli());
+    }
+
+    String getTempTableName(long timestamp) {
```

Comment:
Yeah, I think I mentioned that in some earlier comment, but decided that I would rather move all that work to a separate PR that would refactor the DataAccessImpl / MetricsServiceImpl